packages/core/test/pairing.spec.ts

```diff
@@ -277,4 +277,38 @@ describe("Pairing", () => {
       });
     });
   });
+  describe("events", () => {
+    it("should emit 'pairing_create' event", async () => {
+      let pairingCreatedEvent = false;
+      coreB.pairing.events.on(PAIRING_EVENTS.create, () => (pairingCreatedEvent = true));
+      const { uri } = await coreA.pairing.create();
+      coreB.pairing.pair({ uri });
+      await waitForEvent(() => pairingCreatedEvent);
+    });
+    it("should store pairing before subscribing to its topic", async () => {
+      let pairingCreatedEvent = false;
+      let pairingCreatedEventTime = 0;
+      let subscriptionCreatedEvent = false;
+      let subscriptionCreatedEventTime = 0;
+      const { uri } = await coreA.pairing.create();
+      const { topic } = parseUri(uri);
+      coreB.pairing.events.on(PAIRING_EVENTS.create, () => {
+        expect(coreB.pairing.pairings.keys.length).toBe(1);
```

Comment:
In `vitest`, an `expect` in an event listener throws but you never get to see what went wrong. You just get a timeout here because `waitForEvent` never resolves below. 

I see this is used everywhere in tests so I wonder if this just happens for me :sweat: .

Reply:
you're correct. It does happen to me as well. Moved them out